internal/grpc/services/gateway/usershareprovider.go

```diff
@@ -712,6 +730,55 @@ func (s *svc) removeSpaceShare(ctx context.Context, ref *provider.ResourceId, gr
 	return &collaboration.RemoveShareResponse{Status: status.NewOK(ctx)}, nil
 }
 
+func (s *svc) checkLock(ctx context.Context, shareId *collaboration.ShareId) (*rpc.Status, error) {
+	logger := appctx.GetLogger(ctx)
+	getShareRes, err := s.GetShare(ctx, &collaboration.GetShareRequest{
+		Ref: &collaboration.ShareReference{
+			Spec: &collaboration.ShareReference_Id{Id: shareId},
+		},
+	})
+	if err != nil {
+		msg := "gateway: error calling GetShare"
+		logger.Err(err).Interface("share_id", shareId).Msg(msg)
+		return status.NewInternal(ctx, msg), errors.Wrap(err, msg)
+	}
+	if getShareRes.GetStatus().GetCode() != rpc.Code_CODE_OK {
+		msg := "can not get share stat " + getShareRes.GetStatus().GetMessage()
+		logger.Debug().Interface("share", shareId).Msg(msg)
+		if getShareRes.GetStatus().GetCode() != rpc.Code_CODE_NOT_FOUND {
+			return status.NewNotFound(ctx, msg), errors.New(msg)
+		}
+		return status.NewInternal(ctx, msg), errors.New(msg)
+	}
+	return s.checkShareLock(ctx, getShareRes.Share)
+}
+
+func (s *svc) checkShareLock(ctx context.Context, share *collaboration.Share) (*rpc.Status, error) {
```

Comment:
I think this is tooo restrictive. AFAIU we can
A) allow adding new users to a file
B) increasing the permissions of a share, e.g. add edit permission for a viewer.

The current check disallows any sharing changes when a file is locked.

People often open a file in onlyoffice and then need to ADD new participants.

See https://github.com/owncloud/ocis/issues/8273#issuecomment-1951979302

Reply:
True, we would have to send the lock in the creat/update share opaque... 🤔